### PR TITLE
feat(js,react,api): context HMAC & Inbox dynamic session change fixes NV-6793

### PR DESCRIPTION
**Pull request by ChmaraX**
### What changed? Why was the change needed?

Fixes NV-6809

### Screenshots

https://github.com/user-attachments/assets/dd7fd6c1-47ec-4c4f-9aea-ccbabf1037c5

## Release Notes

* **New Features**
  * Added Context HMAC validation to secure context payloads during session initialization.
  * Introduced new `changeContext()` method to update context with authentication.
  * Added `contextHash` parameter support across session initialization and inbox APIs.

* **Documentation**
  * Updated SDK documentation with Context HMAC setup and usage examples for JavaScript, Next.js, and React.

* **Tests**
  * Added comprehensive test coverage for context HMAC validation scenarios including tampering detection and edge cases.

### CODE REVIEW

File: packages/react/src/hooks/NovuProvider.tsx [hunk 1]
```diff
@@ -70,7 +71,7 @@ export const InternalNovuProvider = (props: NovuProviderProps & { userAgentType:
         defaultSchedule,
         context,
       }),
-    [applicationIdentifier, subscriberHash, backendUrl, apiUrl, socketUrl, useCache, userAgentType, context]
```
Comment (ChmaraX) [inline packages/react/src/hooks/NovuProvider.tsx:73]: There was a bug - when subscriber was dynamically changed in the client code, the session, cache and WS connection were not properly initialized, causing old (previous subscriber) JWT token to be still used.

This useEffect + the useEffect above were creating race condition.

File: packages/js/src/novu.ts [hunk 2]
```diff
@@ -98,6 +99,43 @@ export class Novu implements Pick<NovuEventEmitter, 'on'> {
...
+    });
+
+    // Clear cache and reconnect socket with new token
+    this.notifications.cache.clearAll();
+
+    // Disconnect and reconnect socket to use new JWT token
+    const disconnectResult = await this.socket.disconnect();
+    if (!disconnectResult.error) {
+      await this.socket.connect();
+    }
```
Comment (ChmaraX) [inline packages/js/src/novu.ts:139]: When subscriber (or now context as well) changes dynamically in client's code, we need to properly reinitialise the session, clear cache to not show old subscriber notifications and create socket connection with the correct JWT. 

This was not working before.

**Comment by ChmaraX** [comment]
@coderabbitai summary
